tests/platform_tests/api/test_chassis_fans.py

```diff
@@ -82,17 +75,18 @@ def compare_value_with_platform_facts(self, key, value, fan_idx):
     #
     # Functions to test methods inherited from DeviceBase class
     #
-    def test_get_name(self, duthost, localhost, platform_api_conn):
+    def test_get_name(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, platform_api_conn):
+        duthost = duthosts[enum_rand_one_per_hwsku_hostname]
```

Comment:
The `platform_api_conn` fixture is parameterized using `enum_rand_one_per_hwsku_hostname`. For such test cases that are also parameterized using `enum_rand_one_per_hwsku_hostname`, I am not sure how pytest will generate the eventual list of parameterized test cases. Have you verified this on a chassis? For example, if there are two nodes (node1, node2), will the generated test cases be something like this:
test_get_name[node1](..., platform_api_conn[node1])
test_get_name[node1](..., platform_api_conn[node2])
test_get_name[node2](..., platform_api_conn[node1])
test_get_name[node2](..., platform_api_conn[node2])

Cases like `test_get_name[node1](..., platform_api_conn[node2])
` and `test_get_name[node2](..., platform_api_conn[node1])` would not make sense. For these cases, test executed on node1 would use platform_api_conn for node2. And test executed on node2 would use platform_api_conn for node1.

If pytest just generate test cases `test_get_name[node1](..., platform_api_conn[node1])` and `test_get_name[node1](..., platform_api_conn[node1])`, then it is not an issue.

Can you please confirm?

Reply:
@wangxin I have verified with chassis it is selects same node for test and fixture test_get_name[node1](..., platform_api_conn[node1]), test_get_name[node2](..., platform_api_conn[node2])

Reply:
Thanks for confirming this!